internal/provider/client_ca_cert_resource.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2022 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	sdk_resource "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	clientCACertEndpointTimeout = time.Minute * 10 // TODO: what's a reasonable value here?
+)
+
+type clientCACertResource struct {
+	provider *provider
+}
+
+func NewClientCACertResource() resource.Resource {
+	return &clientCACertResource{}
+}
+
+func (r *clientCACertResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages client CA certs.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Cluster ID",
+			},
+			"x509_pem_cert": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "X509 certificate in PEM format",
+			},
+			"status": schema.StringAttribute{
+				MarkdownDescription: "Status of client CA certs on a cluster",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+func (r *clientCACertResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_client_ca_cert"
+}
+
+func (r *clientCACertResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	var ok bool
+	if r.provider, ok = req.ProviderData.(*provider); !ok {
+		resp.Diagnostics.AddError("Internal provider error",
+			fmt.Sprintf("Error in Configure: expected %T but got %T", provider{}, req.ProviderData))
+	}
+}
+
+func (r *clientCACertResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	// Retrieve values from plan
+	var plan ClientCACertResourceModel
+	diags := req.Config.Get(ctx, &plan) // TODO: why is tfsdk.Config used here instead of tfsdk.Plan
```

Comment:
@erademacher the TF docs/examples use `Plan.Get` here, but I noticed we use `Config.Get` instead?